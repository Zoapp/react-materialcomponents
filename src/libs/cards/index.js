/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
