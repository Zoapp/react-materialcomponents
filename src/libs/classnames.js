/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
