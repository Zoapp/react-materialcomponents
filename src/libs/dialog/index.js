/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
