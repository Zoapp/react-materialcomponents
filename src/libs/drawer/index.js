import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
