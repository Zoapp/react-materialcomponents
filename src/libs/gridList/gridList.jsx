/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Children } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/**
 * mdc-grid-list
 * See:
 * https://material.io/components/web/catalog/grid-lists/
 * https://material-components-web.appspot.com/grid-list.html
 *
 */

const MDC_GRIDLIST = "mdc-grid-list";

const GridList = ({
  children,
  width,
  tileWidth,
  tileBackground,
  gutter,
  headerCaption,
  twolineCaption,
  iconAlign,
  aspectRatio,
  ...props
}) => {
  let classes = MDC_GRIDLIST;
  if (gutter === 1) {
    classes += " mdc-grid-list--tile-gutter-1";
  }
  if (headerCaption) {
    classes += " mdc-grid-list--header-caption";
  }
  if (twolineCaption) {
    classes += " mdc-grid-list--twoline-caption";
  }
  if (iconAlign === "start") {
    classes += " mdc-grid-list--with-icon-align-start";
  } else if (iconAlign === "end") {
    classes += " mdc-grid-list--with-icon-align-end";
  }
  if (aspectRatio) {
    classes += ` mdc-grid-list--tile-aspect-${aspectRatio}`;
  }
  const style = {};
  if (width) {
    style.width = width;
  }
  let ch = children;
  if (tileWidth || tileBackground) {
    ch = Children.map(children, (child) => {
      const ps = { ...child.props };
      if (tileWidth) {
        ps.width = tileWidth;
      }
      if (tileBackground) {
        ps.background = tileBackground;
      }
      return React.cloneElement(child, ps);
    });
  }

  return Rmdc.render((
    <div className={classes}>
      <ul className="mdc-grid-list__tiles" style={style} >
        {ch}
      </ul>
    </div>), props);
};

GridList.defaultProps = {
  mdcElement: MDC_GRIDLIST,
  children: null,
  width: null,
  tileWidth: null,
  tileBackground: null,
  gutter: 0,
  headerCaption: false,
  twolineCaption: false,
  iconAlign: null,
  aspectRatio: null,

};

GridList.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  width: PropTypes.string,
  tileWidth: PropTypes.string,
  tileBackground: PropTypes.string,
  gutter: PropTypes.number,
  headerCaption: PropTypes.bool,
  twolineCaption: PropTypes.bool,
  iconAlign: PropTypes.oneOf(["start", "end"]),
  aspectRatio: PropTypes.oneOf(["1x1", "16x9", "2x3", "3x2", "4x3", "4x3"]),
};

export default GridList;
