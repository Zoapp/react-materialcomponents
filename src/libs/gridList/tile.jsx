/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Children } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";
import Icon from "../components/icon";

/**
 * mdc-grid-tile
 * See:
 * https://material.io/components/web/catalog/grid-lists/
 * https://material-components-web.appspot.com/grid-list.html
 *
 * TODO:
 * - span, order, align
 */

const MDC_GRIDTILE = "mdc-grid-tile";

const Tile = ({
  children, background, width, title, text, icon, ...props
}) => {
  const classes = MDC_GRIDTILE;
  const style = {};
  if (width) {
    style.width = width;
  }
  const primaryStyle = {};
  if (background) {
    primaryStyle.background = background;
  }
  let ch;
  if (children) {
    ch = Children.only(children);
    const c = `mdc-grid-tile__primary-content  ${ch.props.className})`;
    ch = React.cloneElement(ch, { className: c });  
  }
  let secondary;
  if (title) {
    let i;
    if (icon) {
      i = (<Icon className="mdc-grid-tile__icon" name={icon} />);
    }
    let t;
    if (text) {
      t = (<span className="mdc-grid-tile__support-text">{text}</span>);
    }
    secondary = (
      <span className="mdc-grid-tile__secondary">
        {i}
        <span className="mdc-grid-tile__title">{title}</span>
        {t}
      </span>
    );
  }
  return Rmdc.render((
    <ul className={classes} style={style} >
      <div className="mdc-grid-tile__primary" style={primaryStyle} >
        {ch}
      </div>
      {secondary}
    </ul>), props);
};

Tile.defaultProps = {
  mdcElement: MDC_GRIDTILE,
  children: null,
  background: null,
  width: null,
  title: null,
  text: null,
  icon: null,
};

Tile.propTypes = {
  mdcElement: PropTypes.string,
  children: (props, propName) => {
    const prop = props[propName];
    const types = ["img", "div"];
    if (prop && (Children.count(prop) > 1 ||
        types.indexOf(prop.type) === -1)) {
      return new Error("invalid Tile's child");
    }
    return null;
  },
  background: PropTypes.string,
  width: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
  icon: PropTypes.string,
};

export default Tile;
