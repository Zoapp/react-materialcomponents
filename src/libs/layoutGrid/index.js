/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
