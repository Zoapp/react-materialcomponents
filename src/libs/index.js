/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
