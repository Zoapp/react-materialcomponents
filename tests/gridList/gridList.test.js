import React from "react";
import renderer from "react-test-renderer";
import { GridList } from "libs/gridList";

describe("components/GridList", () => {
  it("can have children", () => {
    const tree = renderer
      .create(<GridList>Children are here!</GridList>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a custom width", () => {
    const tree = renderer.create(<GridList width="50px;" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a custom children tile's width", () => {
    const tree = renderer.create(<GridList tileWidth="10px;" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can set background tile's color", () => {
    const tree = renderer.create(<GridList tileBackground="pink" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a gutter", () => {
    const tree = renderer.create(<GridList gutter={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an header caption", () => {
    const tree = renderer.create(<GridList headerCaption />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a two-line caption", () => {
    const tree = renderer.create(<GridList twolineCaption />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can render icon aligned", () => {
    const tree = renderer.create(<GridList iconAlign="start" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can render different aspect ratio", () => {
    const tree = renderer.create(<GridList aspectRatio="16x9" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
