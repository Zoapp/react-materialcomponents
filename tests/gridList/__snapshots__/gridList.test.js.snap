// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GridList can have a variable children tile's width 1`] = `
<div
  className="mdc-grid-list"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can have a variable width 1`] = `
<div
  className="mdc-grid-list"
>
  <ul
    className="mdc-grid-list__tiles"
    style={
      Object {
        "width": "50px;",
      }
    }
  />
</div>
`;

exports[`components/GridList can have an header caption 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--header-caption"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can have an multiline caption 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--twoline-caption"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can have an overflow 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--tile-gutter-1"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can have children 1`] = `
<div
  className="mdc-grid-list"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  >
    Children are here!
  </ul>
</div>
`;

exports[`components/GridList can render different aspect ratio 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--tile-aspect-16x9"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can render icon aligned 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--with-icon-align-start"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;

exports[`components/GridList can set background children tile's color 1`] = `
<div
  className="mdc-grid-list"
>
  <ul
    className="mdc-grid-list__tiles"
    style={Object {}}
  />
</div>
`;
